source/blog/2017-10-10-glimmer-progress-report.md

```diff
@@ -0,0 +1,645 @@
+
+First, a short history lesson. When we introduced components in Ember (back
+before Ember 1.0!), we required them to include a dash (`-`) in their name. This
+rule came from the then-new [Custom Elements spec][custom-elements], a key part
+of the Web Components API.
+
+[custom-elements]: https://www.w3.org/TR/custom-elements/#custom-element-conformance
+
+Web Components have the problem of needing to disambiguate between built-in elements
+and custom elements. What happens if I make a component called `<vr></vr>` and later
+the browser adds a built-in Virtual Reality element with the same name?
+
+The compromise was that custom elements must have a dash, keeping single-word
+elements reserved for future versions of the HTML standard.
+
+Of course, Ember components don't have the same problem, because they use
+`{{`/`}}` as delimeters instead of `<`/`>`. Nonetheless, we preemptively adopted
+this constraint because we assumed Web Components were going to take the world
+by storm and at some point we would need to migrate Ember components to Web
+Components.
+
+As time has passed, though, it's become increasingly clear that the use cases
+served by Web Components, wonderful as they are, do not have the full set of
+functionality to replace everything that an Ember component (or React component,
+etc.) needs to do.
+
+Meanwhile, the more components I write, the more grating the naming restriction
+feels. I hate the cognitive overhead of having to invent silly names like
+`{{x-button}}` when a single word would be much more descriptive.
+
+Unfortunately, this puts us in a bit of a pickle:
+
+1. We want to drop the annoying `dasherized-component` requirement.
+2. We want to adopt `<angle-brackets>` syntax for components, but now we have
+   the same naming collision hazard as Web Components.
+3. People want to use Web Components in their apps, so how do we know if
+   `<my-button>` means "create a custom element" or "create a Glimmer
+   component"?
+
+The core team has circled around different designs for *months*, and this topic
+has dominated both our weekly calls and our in-person meetings.
+
+At the most recent face-to-face meeting in Austin, we finally reached consensus
+on a proposal that I'm really excited about.
+
+How do you disambiguate between Glimmer components and HTML elements? Our
+proposal is to borrow the same rule that React uses: *components always start
+with a capital letter*.
+
+Our above example turns into this:
+
+```hbs
+{{! new Glimmer }}
+<Button @title={{title}} @label={{t "Do Something"}} />
+```
+
+I love this for a few different reasons.
+
+First, to me, the capital letter makes components _really_ stand out in the
+template, and we can improve syntax highlighting in editor plugins to make it
+stand out even more. It also makes it clear when you're invoking a Web Component
+or not, whereas the original Glimmer.js syntax was ambiguous.
+
+Second, for better or worse, many people consider React to be an "industry
```

Comment:
Perhaps there this is item "Third" and "Second" is "single word component names", seems pretty major/notable.